packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1391,6 +1391,25 @@ export class TransactionController extends BaseController<
     return rawTransactions;
   }
 
+  /**
+   * Updates the transaction hash associated with a specific transaction ID.
+   *
+   * @param transactionId - The ID of the transaction to update.
+   * @param hash - The new hash value to be assigned.
+   */
+  updateTransactionHash(transactionId: string, hash: string) {
+    const transactionMeta = this.getTransaction(transactionId);
+    if (!transactionMeta) {
+      log(`Transaction with id ${transactionId} not found.`);
+      return;
+    }
+    transactionMeta.hash = hash;
+    this.updateTransaction(
+      transactionMeta,
+      'TransactionController#updateTransactionHash - Transaction hash updated',
```

Comment:
Nit
```suggestion
      'TransactionController:updateTransactionHash - Transaction hash updated',
```

Sorry I believe it's not mentioned anywhere before, I usually try to put `:` for public methods and `#` in private methods while logging.